proto/kuksa/val/v1/types.proto

```diff
@@ -15,7 +15,24 @@ import "google/protobuf/timestamp.proto";
 // option go_package = "github.com/eclipse/kuksa.val/proto/v1";
 option go_package = "/kuksa_grpc_proto";
 
-message Value {
+// Describes a VSS entry
+// When requesting an entry, the amount of information returned can
+// be controlled by specifying either a `View` or a set of `Field`s.
+message DataEntry {
+  // Defines the full VSS path of the entry.
+  string path = 1;  // [field: FIELD_PATH]
+
+  // The value (datapoint)
+  Datapoint value = 2;  // [field: FIELD_VALUE]
+
+  // Actuator target (only used if the entry is an actuator)
+  Datapoint actuator_target = 3;  // [field: FIELD_ACTUATOR_TARGET]
```

Comment:
Schould this also be optional?

Reply:
It would make sense intuitively, since the field is optional.

However, `optional` only makes sense for primitive types, e.g. `int32`, since the point of having it is to be able to track if the if a field has been set. With proto3, this is not possible for primitive types without adding `optional`.

But since `Datapoint` isn't a primitive type, it's possible to see if the field has been set anyway.

See https://github.com/protocolbuffers/protobuf/blob/v3.12.0/docs/field_presence.md

Reply:
Learned something new (or relearned something old?). Thanks, makes sense!